NXP LPSPI: Fix some regressions

decsny opened the pull request:
Fixes #86148
Fixes #86145

see commit message for detail, fixes fault on ke17z, fixes DT of mcxn236, and fixes high latency regression of repeated spi transfers

Commit:
dts: nxp: mcxn23x: Fix LPSPI fifo sizes

The LPSPI FIFOs on this platform are 8 words long, not 16.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Raymond0225 commented on drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi_common.c:
```diff
@@ -57,22 +62,23 @@ int spi_mcux_configure(const struct device *dev, const struct spi_config *spi_cf
 		}
+
+		base->CR |= LPSPI_CR_RST_MASK;
+		base->CR |= LPSPI_CR_RRF_MASK | LPSPI_CR_RTF_MASK;
```
I think you can remove these lines as they have already initialized in LPSPI_MasterInit

decsny replied:
actually, I just realized, they are still in the wrong place

decsny replied:
@Raymond0225  I updated the PR to do an alternative workaround but I still need to test on S32 hardware, so it will remain on draft still

Commit:
spi_nxp_lpspi: Fix faulting register access

On some platforms, the module is not getting clocked until call to
LPSPI_MasterInit, this will be fixed soon with upcoming update to native
driver and will clock the module in driver init instead of start of
transfer, but for now, move this code within the condition check that
already exists for this purpose.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
spi_nxp_lpspi: Reintroduce fast path no configure

Reintroduce the fast path that skips reconfiguring if we use the same
configuration, this fixes regression that causes a lot of latency at the
start of repeated transfers. Unfortuantely need to find alternative
workaround for S32K3 in order to do this instead of module reset, so
disable skipping for that platform.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>
